### PR TITLE
Fix #6506: Hide internal Tuple.fromArray implementation

### LINKED ISSUE #6506: Tuple.fromArray fails if not given concrete tuple type parameter
```scala
 val t: Tuple = Tuple.fromArray(Array[Object]("1", "2", "3"))
```
fails with 
```scala
-- Error: /Users/nicolasstucki/GitHub/dotty/library/src-3.x/scala/Tuple.scala:168:21 
168 |    inline constValue[BoundedSize[T]] match {
    |           ^^^^^^^^^^^^^^^^^^^^^^^^^^
    |not a constant type: scala.Tuple.BoundedSize[Nothing]; cannot take constValue
    | This location is in code that was inlined at Foo.scala:2
```

### PR DESCRIPTION
**Pull request by nicolasstucki**
Renamed old `fromArray` to `knowTupleFromArray` and added a `fromArray` that a user can call.

### CODE REVIEW

**Comment by milessabin** [comment]
A "fromProduct" would also be useful :-)

**Comment by milessabin** [comment]
Also from/to nested pairs ... but maybe I should just do these myself ;-)

**Comment by nicolasstucki** [comment]
I added the `fromProduct` in #6514

File: library/src-3.x/scala/Tuple.scala [hunk 1]
```diff
@@ -47,7 +47,7 @@ sealed trait Tuple extends Any {
         val t = asInstanceOf[Tuple4[_, _, _, _]]
         Tuple5(x, t._1, t._2, t._3, t._4).asInstanceOf[Result]
       case Some(n) =>
-        fromArray[H *: this.type](cons$Array(x, toArray))
+        knowTupleFromArray[H *: this.type](cons$Array(x, toArray))
```
Comment (odersky) [inline library/src-3.x/scala/Tuple.scala]: knownTupleFromArray

**Pull request merged**